lib/spack/spack/test/cmd/env.py

```diff
@@ -760,7 +760,7 @@ def packages_file(tmpdir):
     raw_yaml = """
 packages:
   mpileaks:
-    version: [2.2]
+    version: ["2.2"]
```

Comment:
ah ok I see where we were leaking floats into the version constructor.

Reply:
Oof yeah. I think this actually warrants a config `update(...)` where `int/float` -> `str`. Pretty sure it will confuse folks...

Reply:
Yeah, it should really be int, or str required.  The one test package.py file that had explicit floats was fixed too right?